stan/math/fwd/fun/hypergeometric_pFq.hpp

```diff
@@ -22,32 +22,42 @@ namespace math {
  * @return Generalized hypergeometric function
  */
 template <typename Ta, typename Tb, typename Tz,
-          require_all_matrix_t<Ta, Tb>* = nullptr,
-          require_return_type_t<is_fvar, Ta, Tb, Tz>* = nullptr>
-inline return_type_t<Ta, Tb, Tz> hypergeometric_pFq(const Ta& a, const Tb& b,
-                                                    const Tz& z) {
-  using fvar_t = return_type_t<Ta, Tb, Tz>;
-  ref_type_t<Ta> a_ref = a;
-  ref_type_t<Tb> b_ref = b;
-  auto grad_tuple = grad_pFq(a_ref, b_ref, z);
-
-  typename fvar_t::Scalar grad = 0;
-
-  if (!is_constant<Ta>::value) {
-    grad += dot_product(forward_as<promote_scalar_t<fvar_t, Ta>>(a_ref).d(),
-                        std::get<0>(grad_tuple));
+          typename FvarT = return_type_t<Ta, Tb, Tz>,
+          bool grad_a = !is_constant<Ta>::value,
+          bool grad_b = !is_constant<Tb>::value,
+          bool grad_z = !is_constant<Tz>::value,
+          require_all_vector_t<Ta, Tb>* = nullptr,
+          require_return_type_t<is_fvar, FvarT>* = nullptr>
```

Comment:
`FvarT` is already defined as `typename FvarT = return_type_t<Ta, Tb, Tz>`. 

Should this be simplified to `require_fvar_t<FvarT>`?